pkg/pillar/pubsub/updaters.go

```diff
@@ -42,6 +43,5 @@ func (u *Updaters) Remove(updater chan Notify) {
 	if !found {
 		log.Fatal("updaters.remove(): not found\n")
 	}
-	u.servers = servers
```

Comment:
Doesn't this stop the `Remove()` update?

Reply:
No, its updated in place. Basically the append makes it move the entries post the removed up one.

Reply:
That doesn't seem right to me. We create a new slice from the old slice, and then append all except for found from the old slice. But `servers` and `u.servers` are 2 distinct slices. But even if it does, we will end up double adding.  See https://play.golang.org/p/5S94Zquhnpb

Reply:
Sorry my bad. Too subtle. Need to add that line back.